pipenv/vendor/requirementslib/models/setup_info.py

```diff
@@ -1141,29 +1137,15 @@ def run_setup(script_path, egg_base=None):
         sys.path.insert(0, target_cwd)
 
         save_argv = sys.argv.copy()
-        try:
-            global _setup_distribution, _setup_stop_after
-            _setup_stop_after = "run"
-            sys.argv[0] = script_name
-            sys.argv[1:] = args
-            with open(script_name, "rb") as f:
-                contents = f.read().replace(rb"\r\n", rb"\n")
-                exec(contents, g)
-        # We couldn't import everything needed to run setup
-        except Exception:
-            python = os.environ.get("PIP_PYTHON_PATH", sys.executable)
+        python = os.environ.get("PIP_PYTHON_PATH", sys.executable)
 
-            sp.run(
-                [python, "setup.py"] + args,
-                cwd=target_cwd,
-                stdout=sp.PIPE,
-                stderr=sp.PIPE,
-            )
-        finally:
-            _setup_stop_after = None
-            sys.argv = save_argv
-            _setup_distribution = get_metadata(egg_base, metadata_type="egg")
-        dist = _setup_distribution
+        sp.run(
+            [python, "setup.py"] + args,
+            cwd=target_cwd,
+            stdout=sp.PIPE,
+            stderr=sp.PIPE,
```

Comment:
```suggestion
            capture_output=True
```
---
% `MY_CODE="import subprocess as sp ; sp.run([python, "setup.py"] + args, cwd=target_cwd, stdout=sp.PIPE, stderr=sp.PIPE)"`

% `echo $MY_CODE | ruff --select=UP022 --fix -`
```
import subprocess as sp ; sp.run([python, setup.py] + args, cwd=target_cwd, capture_output=True)
```
% `ruff rule UP022`
# replace-stdout-stderr (UP022)

Derived from the **pyupgrade** linter.

Autofix is always available.

## What it does
Checks for uses of `subprocess.run` that send `stdout` and `stderr` to a
pipe.

## Why is this bad?
As of Python 3.7, `subprocess.run` has a `capture_output` keyword argument
that can be set to `True` to capture `stdout` and `stderr` outputs. This is
equivalent to setting `stdout` and `stderr` to `subprocess.PIPE`, but is
more explicit and readable.

## Example
```python
import subprocess

subprocess.run(["foo"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Use instead:
```python
import subprocess

subprocess.run(["foo"], capture_output=True)
```

## References
- [Python 3.7 release notes](https://docs.python.org/3/whatsnew/3.7.html#subprocess)
- [Python documentation: `subprocess.run`](https://docs.python.org/3/library/subprocess.html#subprocess.run)

Reply:
Resolved in #5762

Reply:
Was the removal of `cwd=target_cwd` intentional?

Reply:
@cclauss intentional, as it seemed unnecessary with the `cd(target_cwd)` context manager -- should already be in that directory.

Reply:
Well its finally having a good run -- some just turned green.